pynestml/transformers/synapse_post_neuron_transformer.py

```diff
@@ -277,6 +276,15 @@ def transform_neuron_synapse_pair_(self, neuron, synapse):
                                              for neuron_state_var in syn_to_neuron_state_vars
                                              if new_synapse.get_kernel_by_name(neuron_state_var) is None]
 
+        # all state variables that will be moved from synapse to neuron
+        syn_to_neuron_state_vars = []
```

Comment:
The same variable is assigned in line 267. Why is it initialized here again?

Reply:
It's first assigned the variables from an initial scan of the model, then a recursive scan is done to find dependent variables, then it gets overwritten in a way that includes the newly found variables.